contracts/tokenomics/maker/src/contract.rs

```diff
@@ -430,15 +462,15 @@ fn distribute_astro(deps: DepsMut, env: Env, info: MessageInfo) -> Result<Respon
         return Err(ContractError::Unauthorized {});
     }
 
-    let cfg = CONFIG.load(deps.storage)?;
+    let mut cfg = CONFIG.load(deps.storage)?;
     let astro = token_asset_info(cfg.astro_token_contract.clone());
 
-    let balance = astro.query_pool(&deps.querier, env.contract.address)?;
+    let balance = astro.query_pool(&deps.querier, env.contract.address.clone())?;
```

Comment:
Suggest moving this call into distribute function

Reply:
`distribute` is also used in `collect`. Should we change `collect` method as well?

Reply:
yes, that's the reason of moving so we have it DRY